lib/scan.ts

```diff
@@ -10,6 +10,13 @@ import { fromUrl } from 'hosted-git-info';
 import { getSignature } from './signatures';
 import { getTarget } from './git';
 
+function getSignatureMapper(filePath: string): Promise<SignatureResult | null> {
+  // Currently "dubhash" is the default hashType
+  // To generate "uhash" signatures, change the below line to:
+  // return getSignature(filePath, 'uhash');
+  return getSignature(filePath);
+}
```

Comment:
The way that we are dealing with p-map and mapper here will not allow us of easily switch from an algorithm to another if we pretend to add an experimental flag e.g. `--uhash`.

So I will be adding few suggestions